src/main/java/com/epam/processor/DataProcessor.java

```diff
@@ -93,31 +166,62 @@ public RoadAccident getAccidentByIndex(String index){
      * @return
      */
     public Collection<RoadAccident> getAccidentsByLocation(float minLongitude, float maxLongitude, float minLatitude, float maxLatitude){
-        return null;
+        return roadAccidentList.stream()
+                .filter(r -> r.getLongitude()>=minLongitude && r.getLongitude()<=maxLongitude
+                        && r.getLatitude()>=minLatitude && r.getLatitude()<=maxLatitude )
+                .collect(Collectors.toList());
     }
 
     /**
      * find the weather conditions which caused max number of incidents
      * @return
      */
     public List<String> getTopThreeWeatherCondition(){
-        return null;
+        Map<String, Long> weatherCondition = roadAccidentList.stream()
+                .collect(Collectors.groupingBy(RoadAccident::getWeatherConditions,
+                        Collectors.counting()));
+//        System.out.println(weatherCondition);
+
+        Ordering<Map.Entry<String,Long>> ordering = new Ordering<Map.Entry<String,Long>>() {
```

Comment:
That's pretty cool! I didn't know about this class Ordering. Thanks for discovering it :)